src/main/java/graphql/nadel/engine/FieldMetadataUtil.java

```diff
@@ -135,6 +138,20 @@ private static String writeMetadata(List<FieldMetadata> fieldMetadata) {
         }
     }
 
+    private static Field writeMetadata(Field field, List<FieldMetadata> newMetadata) {
+        return field.transform(builder -> writeMetadata(builder, newMetadata));
+    }
+
+    private static Field.Builder writeMetadata(Field.Builder builder, List<FieldMetadata> newMetadata) {
+        String newSerializedValue = writeMetadata(newMetadata);
+        return builder.additionalData(NADEL_FIELD_METADATA, newSerializedValue);
+    }
+
+    private static List<FieldMetadata> readMetadata(Field field) {
```

Comment:
it is a good practice never to return `null` from a method with List/Collection. This simplifies the client code.